metaserver/src/test/scala/tests/compiler/SignatureHelpTest.scala

```diff
@@ -199,19 +247,33 @@ object SignatureHelpTest extends CompilerSuite {
     """.stripMargin,
     """
       |{
-      |  "signatures" : [ {
-      |    "label" : "<init>(name: String)User",
-      |    "parameters" : [ {
-      |      "label" : "name: String"
-      |    } ]
-      |  }, {
-      |    "label" : "<init>(name: String, age: Int)User",
-      |    "parameters" : [ {
-      |      "label" : "name: String"
-      |    }, {
-      |      "label" : "age: Int"
-      |    } ]
-      |  } ],
+      |  "signatures" : [
+      |    {
+      |      "label" : "<init>(name: String)User",
+      |      "documentation" : null,
```

Comment:
Any concern that the previous json library did not serialization null values and circe appears presently configured to serialization them as JsNull?

Reply:
From a protocol POV it should be irrelevant.
As mentioned in https://github.com/scalameta/language-server/pull/165#commitcomment-26641371 I treated this as an implementation detail. Changing the default pretty-printer would've been much more bothersome.

Reply:
Printer is easy to change, IIRC it’s just printer.copy(dropNullValues = true)
I don’t think changing json libraries should require test changes.  Are we comparing raw strings where we should be comparing parsed json in tests?

Reply:
> Printer is easy to change, IIRC it’s just printer.copy(dropNullValues = true)

It is, but then you have to pass around the printer instance and do `json.pretty(printer)` instead of `json.spaces2`.

> Are we comparing raw strings where we should be comparing parsed json in test

Using directly the json strings was a choice we took a while ago and it served us well so far.

I would be ok in comparing with parsed json instead. Something like:

``
import io.circe.Json.{obj, arr, fromInt => int, fromString => str, False}
...

check(
  "foo",
  obj(
    "isIncomplete" -> False,
    "items" -> arr(
      obj(
        "label" -> str(label),
        "kind" -> int(kind.value),
        "detail" -> str(detail),
        "sortText" -> str("00000")
      )
    )
  )
)
```        

wdyt @olafurpg?

Reply:
What do we gain by moving to comparing parsed json instead of strings? I personally think strings are nice because

1. test errors are reported as unified diff which are easy to read
2. false negatives are easy to fix by copy-pasting the failed test output and pasting it into the field

Reply:
For the tests I would recommend using `dropNullValues = true` to minimize the string output, I have no preference for the messages sent from MessageWriter

Reply:
> I don’t think changing json libraries should require test changes. 

I think it's totally fine if the pretty-printers are incompatible. It would be nice if there was a compatible printer so we can keep the tests unchanged, I personally liked the play-json pretty-printer output since it was quite compact compared to the output in these tests.